eZ/Publish/SPI/Search/Indexing.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\Search;
+
+/**
+ * Interface for indexing in search backend.
+ */
+interface Indexing
+{
+    /**
+     * Purge search engine index.
+     */
+    public function purgeIndex();
+
+    /**
+     * Make indexing operation permanent if Search Handler needs and supports it.
+     * Note: it is a responsibility of a Search Handler to decide if this is needed and how to do this.
+     */
+    public function commitIndex();
```

Comment:
The reason I introduced here the new method instead of reusing existing in Solr and LSE `commit` method is BC. `commit` has one optional parameter `$flush` which for generic purpose makes no sense. Solr Search Handler itself when implementing `commitIndex` should decide if this should be hard or soft commit. LSE implementation of this method is empty.